src/promptflow-azure/tests/sdk_cli_azure_test/e2etests/test_telemetry.py

```diff
@@ -466,3 +466,59 @@ def check_evelope():
             )
             logger.handlers[0].flush()
             check_evelope()
+
+    @pytest.mark.skipif(
+        condition=not pytest.is_live,
+        reason="Live mode can run successfully, but an error will be reported when recording.",
+    )
+    def test_flow_type_with_pfazure_flows(self, pf, randstr: Callable[[str], str]):
+        from promptflow._constants import FlowType
+        from promptflow._sdk._configuration import Configuration
+        from promptflow._sdk._telemetry.logging_handler import PromptFlowSDKExporter
+
+        envelope = None
+        flow_type = None
+        config = Configuration.get_instance()
+        custom_dimensions = {
+            "python_version": platform.python_version(),
+            "installation_id": config.get_or_set_installation_id(),
+        }
+        log_to_envelope = PromptFlowSDKExporter(
+            connection_string="InstrumentationKey=00000000-0000-0000-0000-000000000000",
+            custom_dimensions=custom_dimensions,
+        )._log_to_envelope
+
+        def log_event(log_data):
+            nonlocal envelope
+            envelope = log_to_envelope(log_data)
+
+        def check_evelope():
+            assert envelope.data.base_data.name.startswith("pfazure.flows.create_or_update")
+            custom_dimensions = pydash.get(envelope, "data.base_data.properties")
+            assert isinstance(custom_dimensions, dict)
+            assert "flow_type" in custom_dimensions
+            assert custom_dimensions["flow_type"] == flow_type
+
+        with patch.object(PromptFlowSDKExporter, "_log_to_envelope", side_effect=log_event), patch(
+            "promptflow._sdk._telemetry.telemetry.get_telemetry_logger", side_effect=get_telemetry_logger
+        ):
+            flow_type = FlowType.DAG_FLOW
+            try:
+                pf.flows.create_or_update(
+                    flow=FLOWS_DIR / "print_input_flow",
```

Comment:
what if user passed an invalid type obj to flow?

Reply:
The flow type field will not be added:
![image](https://github.com/microsoft/promptflow/assets/23182548/0ec1e956-da9a-4361-a25d-cd6e552712b9)